test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
```

Comment:
Something I don't understand is can multiple top level nodes exist next to each other?
Is there or should there be some class that knows and manages all nodes created in the application? like a `NodeManager` that would have methods like `getTopLevelNodes` or `getNodeById` methods?

A good way to split up responsibilities is to have this class consist of only how the data is modeled and a manager or service to use these nodes and manipulate them including moving methods like `add_child` over to the manager.